py/visdom/__init__.py

```diff
@@ -1073,6 +1078,17 @@ def embeddings(self, features, labels, data_getter=None, data_type=None,
         _title2str(opts)
         _assert_opts(opts)
 
+        loading_message = {
```

Comment:
You _could_ add almost this exact setup before the `do_tsne` call in the `elif event['event_type'] == 'RegionSelected':` branch to make this work with region selected as well. (the primary difference being sending the already existing data but setting `isLoading`. Ultimately this isn't particularly efficient though, I think long term we're going to need to support the same kind of event system that we have in the reverse direction into the forward direction. Might be worth tabling the want to have a loading indicator for selecting regions until we work out what that might look like.

Reply:
I tried mirroring the setup but was getting a cryptic error. Will show you in person.

Reply:
Discussed offline to table this for now. Marking as resolved.